client/js/components/EditProfileForm/EditProfileForm.js

```diff
@@ -0,0 +1,199 @@
+import { Form, Field } from "react-final-form";
+import {
+  View,
+  Image,
+  TouchableOpacity,
+  TextInput,
+  Text,
+  Alert
+} from "react-native";
+import styles from "./styles";
+import PropTypes from "prop-types";
+
+class EditProfileForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      newPassword: "",
+      verifyPassword: "",
+      firstname: this.props.data ? this.props.data.firstname : "First name",
+      lastname: this.props.data ? this.props.data.lastname : "Last name",
+      email: this.props.data ? this.props.data.email : "Email"
+    };
+  }
+
+  render() {
+    const { updateUser, id, navigation } = this.props;
+
+    return (
+      <View style={styles.editprofilewrapper}>
+        <React.Fragment>
+          <Image
+            source={require("../../assets/images/placeholder.jpg")}
+            style={styles.avatar}
+          />
+          <TouchableOpacity style={styles.profileEditButton}>
+            <Text>EDIT </Text>
+          </TouchableOpacity>
+        </React.Fragment>
+
+        <Form
+          onSubmit={async () => {
+            try {
+              await updateUser({
+                variables: {
+                  id: id,
+                  firstname: this.state.firstname,
+                  lastname: this.state.lastname,
+                  email: this.state.email,
+
+                  // check if `new password` and `verify password` fields match. If yes, proceed with mutation.
+                  password:
+                    this.state.newPassword === this.state.verifyPassword
+                      ? this.state.verifyPassword
+                      : Alert.alert(
+                          "Your new password doesn't match",
+                          "Please re-enter your new password",
+                          [{ text: "Got it" }]
+                        )
+                }
+              });
+              //once mutation goes through successfully, feedback to the user:
+              Alert.alert(
+                "Done!",
+                "Your information was updated successfully.",
+                [
+                  {
+                    text: "Go Back",
+                    onPress: () => navigation.navigate("Profile")
+                  }
+                ]
+              );
+            } catch (e) {
+              return <Text>{e}</Text>;
+            }
+          }}
+        >
+          {({ handleSubmit, values }) => (
+            <View>
+              <View style={styles.formWrapper}>
+                <Field name="firstname">
+                  {({ input }) => (
+                    <TextInput
+                      {...input}
+                      style={styles.editProfileInputs}
+                      placeholder={this.state.firstname}
+                      placeholderTextColor="black"
+                      type="text"
+                      onChangeText={firstname => this.setState({ firstname })}
+                    />
```

Comment:
we do not use component state with the final form since the final form has its own state. Make the changes, please. Also, you should validate your user's input and show appropriate errors when needed.